kotlin-sdk-server/src/commonMain/kotlin/io/modelcontextprotocol/kotlin/sdk/server/StreamableHttpServerTransport.kt

```diff
@@ -0,0 +1,575 @@
+                reject(HttpStatusCode.MethodNotAllowed, ErrorCode.Unknown(-32000), "Method not allowed.")
+            }
+        }
+    }
+
+    /**
+     * Handles POST requests containing JSON-RPC messages
+     */
+    public suspend fun handlePostRequest(session: ServerSSESession?, call: ApplicationCall) {
+        try {
+            if (!enableJsonResponse && session == null) error("Server session can't be null with json response")
+
+            val acceptHeader = call.request.header(HttpHeaders.Accept)
+            val isAcceptEventStream = acceptHeader.accepts(ContentType.Text.EventStream)
+            val isAcceptJson = acceptHeader.accepts(ContentType.Application.Json)
+
+            if (!isAcceptEventStream || !isAcceptJson) {
+                call.reject(
+                    HttpStatusCode.NotAcceptable,
+                    ErrorCode.Unknown(-32000),
+                    "Not Acceptable: Client must accept both application/json and text/event-stream",
+                )
+                return
+            }
+
+            if (!call.request.contentType().match(ContentType.Application.Json)) {
+                call.reject(
+                    HttpStatusCode.UnsupportedMediaType,
+                    ErrorCode.Unknown(-32000),
+                    "Unsupported Media Type: Content-Type must be application/json",
+                )
+                return
+            }
+
+            val messages = parseBody(call) ?: return
+            val isInitializationRequest = messages.any {
+                it is JSONRPCRequest && it.method == Method.Defined.Initialize.value
+            }
+
+            if (isInitializationRequest) {
+                if (initialized.load() && sessionId != null) {
+                    call.reject(
+                        HttpStatusCode.BadRequest,
+                        ErrorCode.Defined.InvalidRequest,
+                        "Invalid Request: Server already initialized",
+                    )
+                    return
+                }
+                if (messages.size > 1) {
+                    call.reject(
+                        HttpStatusCode.BadRequest,
+                        ErrorCode.Defined.InvalidRequest,
+                        "Invalid Request: Only one initialization request is allowed",
+                    )
+                    return
+                }
+
+                sessionMutex.withLock {
+                    if (sessionId != null) return@withLock
+                    sessionId = sessionIdGenerator?.invoke()
+                    initialized.store(true)
+                    sessionId?.let { onSessionInitialized?.invoke(it) }
+                }
+            } else {
+                if (!validateSession(call) || !validateProtocolVersion(call)) return
+            }
+
+            val hasRequest = messages.any { it is JSONRPCRequest }
+            if (!hasRequest) {
+                call.respondNullable(status = HttpStatusCode.Accepted, message = null)
```

Comment:
Perhaps because of the `Accept` request header, this line returns `406 Not Acceptable` instead of `202`. MCP Inspector throws an error in log, and the Python client crashes because of this. For me, `call.respondBytes(status = HttpStatusCode.Accepted, bytes = ByteArray(0))` helped.